frappe/public/js/frappe/form/quick_entry.js

```diff
@@ -191,7 +191,10 @@ frappe.ui.form.QuickEntryForm = class QuickEntryForm {
 					doc: me.dialog.doc,
 				},
 				callback: function (r) {
-					if (frappe.model.is_submittable(me.doctype)) {
+					if (
+						frappe.model.is_submittable(me.doctype) &&
```

Comment:
```suggestion
						frappe.model.can_submit(me.doctype) &&
```

Maybe we should take it a bit further. Because submit option pops up and on clicking an insufficient perms error is thrown.